discourse-doctor

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env bash
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $DIR
+
+LOG_FILE="/tmp/discourse-debug.txt"
+rm -f $LOG_FILE
+echo DISCOURSE DOCTOR `date` | tee -a $LOG_FILE
+
+##
+## Make sure only root can run our script
+##
+check_root() {
+  if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root. Please sudo or log in as root first." 1>&2
+    exit 1
+  fi
+}
+
+##
+## Check whether a connection to HOSTNAME ($1) on PORT ($2) is possible
+##
+connect_to_port () {
+  HOST="$1"
+  PORT="$2"
+  VERIFY=`date +%s | sha256sum | base64 | head -c 20`
+  echo -e "HTTP/1.1 200 OK\n\n $VERIFY" | nc -w 4 -l -p $PORT >/dev/null 2>&1 &
+  if curl --proto =http -s $HOST:$PORT --connect-timeout 3 | grep $VERIFY
+  then
+    echo "OK"
+    return 0
+  else
+    echo "NO K"
+    return 1
+  fi
+}
+
+check_ip_match () {
+  HOST="$1"
+  echo
+  echo Checking your domain name . . . | tee -a $LOG_FILE
```

Comment:
Could we have generic logging function?

So we would have
```
log("Checking your domain name ...")
```

instead of

```
echo Checking your domain name . . . | tee -a $LOG_FILE
```

This would remove a lot of redundant code.